### PR TITLE
Factory bot and rubocop

### PR DESCRIPTION
**Pull request by tylertomlinson**
Law abiding citizens now. `Rubocop` 

Implemented Factory bot for:
- `User`
- `Post`
- `Fish`

Might need to change the an attribute  on factory bot since `Faker` is pulling from a limited pool and will randomly fail. 

Screen Shot 2020-09-02 at 5 32 20 PM

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -1,5 +1,31 @@
-@user = User.create!(name: "Chase", city: 'Denver', state: "Colorado", email: 'lethargicgarfield@gmail.com', password: "password_mania")
-@post1 = @user.posts.create!(title: "Cutthroat on the 4th!", city: "Silverthorne", state: "CO", body_of_water: "Blue River", region: 'Central Colorado', fly_or_conventional: "Fly", privacy: "Public")
-@rainbow = @post1.fish.create!(species: 'Rainbow Trout', harvested: false, length: 15, quantity: 1, photo: 'rainbow_blue_river.jpg')
-@post2 = @user.posts.create!(title: "Dream Stream Cuttbow", city: "Fairplay", state: "CO", fly_or_conventional: "Fly", body_of_water: "Dream Stream", region: 'South Park', privacy: "Public")
-@cuttbow = @post2.fish.create!(species: 'Hybrid Trout', harvested: false, length: 18, quantity: 1, photo: 'cuttbow.jpg')
+@user = User.create!(name: 'Chase',
+                     city: 'Denver',
+                     state: 'Colorado',
+                     email: 'lethargicgarfield@gmail.com',
+                     password: 'password_mania')
```
Comment (tylertomlinson) [inline db/seeds.rb:5]: Idk what this is about. Looks fine on my editor

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  Enabled: false
+
+Style/Documentation:
+  Enabled: false
+
+Style/FrozenStringLiteralComment:
+  Enabled: false
+
+Metrics/BlockLength:
+  Max: 62
```
Comment (cgaddis36) [inline .rubocop.yml:25]: I dig it, nice addition of rubocop

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
       redirect_to '/login'
     end
   end
 
   def destroy
     session.delete(:user_id)
-    flash[:notice] = "Successfully Logged Out"
-    redirect_to "/"
+    flash[:notice] = 'Successfully Logged Out'
+    redirect_to '/'
```
Comment (cgaddis36) [inline app/controllers/sessions_controller.rb:22]: thanks rubo!

File: .rubocop.yml (hunk 2 above)
Comment (tylertomlinson) [reply .rubocop.yml:25]: We law abiding homie!

File: app/models/fish.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class Fish < ApplicationRecord
-  validates_presence_of :quantity
-  validates_presence_of :species
+  validates :quantity, presence: true
+  validates :species, presence: true
```
Comment (cgaddis36) [inline app/models/fish.rb:4]: I need to get used to this new syntax

File: app/models/user.rb [hunk 5]
```diff
@@ -14,7 +15,7 @@ def self.update_or_create(auth_info)
                         email: auth_info[:info][:email],
                         name: auth_info[:info][:name],
                         google_token: auth_info[:credentials][:token],
-                        password: "#{SecureRandom.hex(8)}" }
+                        password: SecureRandom.hex(8).to_s }
     user.save!
```
Comment (cgaddis36) [inline app/models/user.rb:19]: this is much cleaner

File: db/seeds.rb (hunk 1 above)
Comment (cgaddis36) [reply db/seeds.rb:5]: probably an indentation issue from rubo

File: spec/factories/fish.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+  factory :fish do
+    species { Faker::Movies::StarWars.specie }
```
Comment (cgaddis36) [inline spec/factories/fish.rb:3]: love these star wars factories

File: spec/features/user/post/new_spec.rb [hunk 7]
```diff
@@ -2,9 +2,10 @@
 
 RSpec.describe 'As a logged in user' do
   before(:each) do
-    @user = User.create!(name: "Carl", city: 'Denver', state: "Colorado", email: 'lethargicgarfield@gmail.com', password: "password_mania")
+    @user = create(:user)
```
Comment (cgaddis36) [inline spec/features/user/post/new_spec.rb:5]: so much nicer

Comment (tylertomlinson) [reply spec/features/user/post/new_spec.rb:5]: Yes. Factory bot ftw

**Review by cgaddis36 — APPROVED** [review]
Looks good, thanks for adding in those Factories and breaking apart the registration sad path into its own separate test

**Pull request merged**